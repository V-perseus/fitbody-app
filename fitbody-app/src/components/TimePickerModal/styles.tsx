import { StyleSheet } from 'react-native'
import globals from '../../config/globals'

export default StyleSheet.create({
  date: {
    fontFamily: globals.fonts.secondary.style.fontFamily,
    color: 'black',
    fontSize: 24,
    marginTop: 10,
    marginBottom: 10,
  },
  container: {
    padding: 50,
    paddingTop: 20,
    paddingBottom: 0,
    width: globals.window.width * 0.9,
    height: 435.1,
    backgroundColor: globals.styles.colors.colorWhite,
    borderRadius: 10,
  },
  pickerContainer: { flex: 1, flexDirection: 'row', justifyContent: 'center', alignItems: 'center' },
  hoursPickerContainer: { width: 60 },
  minutesPickerContainer: { width: 60 },
  ampmPickerContainer: { width: 50 },
  gradientTop: { position: 'absolute', borderWidth: 0, borderRadius: 10, top: 20, width: globals.window.width * 0.9, height: 230 },
  gradientBottom: { position: 'absolute', borderWidth: 0, borderRadius: 10, bottom: 74, width: globals.window.width * 0.9, height: 230 },
  selectorWindow: {
    position: 'absolute',
    top: 165.5,
    borderColor: globals.styles.colors.colorGray,
    width: globals.window.width * 0.9,
    height: 50,
    borderBottomWidth: 1,
    borderTopWidth: 1,
  },
  selectButton: { width: globals.window.width * 0.8 },
})
