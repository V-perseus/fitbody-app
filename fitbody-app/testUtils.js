import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
