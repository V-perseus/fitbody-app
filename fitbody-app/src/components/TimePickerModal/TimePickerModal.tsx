import React, { useEffect, useState } from 'react'
import { View, Text } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import Modal from 'react-native-modal'
import ScrollPicker from 'react-native-wheel-scrollview-picker'
import moment from 'moment'

import styles from './styles'
import globals from '../../config/globals'

import CloseIcon from '../../../assets/images/svg/icon/24px/close.svg'
import PinkButton from '../PinkButton'
import { ButtonOpacity } from '../Buttons/ButtonOpacity'

const HOURS = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']
const MINUTES = ['00', '30']
const AMPM = ['AM', 'PM']

const NO_OP = () => {}

interface ITimePickerModalProps {
  testID?: string
  visible: boolean
  onClose: () => void
  onChange: (time: string) => void
}
const TimePickerModal: React.FC<ITimePickerModalProps> = ({ testID, visible, onClose = NO_OP, onChange = NO_OP }) => {
  const [isModalOpen, setIsModalOpen] = useState(visible)
  const [activeHourIndex, setActiveHourIndex] = useState(6)
  const [activeMinuteIndex, setActiveMinuteIndex] = useState(0)
  const [activeAmpmIndex, setActiveAmpmIndex] = useState(0)

  useEffect(() => {
    setIsModalOpen(visible)
  }, [visible])

  function handleClose() {
    setIsModalOpen(false)
    onClose()
  }

  function handleSelectTime() {
    const time = `${HOURS[activeHourIndex]}:${MINUTES[activeMinuteIndex]} ${AMPM[activeAmpmIndex]}`
    setIsModalOpen(false)
    onChange(moment(time, 'hh:mm a').format('HH:mm'))
  }

  function renderHoursItem(data: string | number) {
    return (
      <View>
        <Text style={[styles.date, { textAlign: 'right' }]}>{data}</Text>
      </View>
    )
  }

  function renderMinutesItem(data: string | number) {
    return (
      <View>
        <Text style={styles.date}>{data}</Text>
      </View>
    )
  }

  function renderAmpmItem(data: string | number) {
    return (
      <View>
        <Text style={styles.date}>{data}</Text>
      </View>
    )
  }

  return (
    <Modal
      onBackdropPress={handleClose}
      onBackButtonPress={handleClose}
      backdropOpacity={0.5}
      useNativeDriver={true}
      animationIn="fadeIn"
      animationOut="fadeOut"
      isVisible={isModalOpen}>
      {isModalOpen ? (
        <View testID={testID} style={styles.container}>
          <View style={styles.pickerContainer}>
            <View style={styles.hoursPickerContainer}>
              <ScrollPicker
                dataSource={HOURS}
                selectedIndex={activeHourIndex}
                itemHeight={49.3}
                wrapperHeight={345}
                highlightColor={globals.styles.colors.colorWhite}
                renderItem={renderHoursItem}
                onValueChange={(data, selectedIndex) => {
                  setActiveHourIndex(selectedIndex)
                }}
              />
            </View>
            <Text style={[styles.date, { textAlign: 'right', marginTop: 22 }]}>:</Text>
            <View style={styles.minutesPickerContainer}>
              <ScrollPicker
                dataSource={MINUTES}
                selectedIndex={activeMinuteIndex}
                itemHeight={49.3}
                wrapperHeight={345}
                highlightColor={globals.styles.colors.colorWhite}
                renderItem={renderMinutesItem}
                onValueChange={(data, selectedIndex) => {
                  setActiveMinuteIndex(selectedIndex)
                }}
              />
            </View>
            <View style={styles.ampmPickerContainer}>
              <ScrollPicker
                dataSource={AMPM}
                selectedIndex={activeAmpmIndex}
                itemHeight={49.3}
                wrapperHeight={345}
                highlightColor={globals.styles.colors.colorWhite}
                renderItem={renderAmpmItem}
                onValueChange={(data, selectedIndex) => {
                  setActiveAmpmIndex(selectedIndex)
                }}
              />
            </View>
          </View>
          {/* Top fade out effect */}
          <LinearGradient style={styles.gradientTop} colors={['rgba(255, 255, 255, 1)', 'rgba(255, 255, 255, 0)']} pointerEvents={'none'} />
          {/* Bottom fade out effect */}
          <LinearGradient
            style={styles.gradientBottom}
            colors={['rgba(255, 255, 255, 0)', 'rgba(255, 255, 255, 1)']}
            pointerEvents={'none'}
          />
          <ButtonOpacity testID="close_btn" onPress={handleClose} style={{ padding: 15, position: 'absolute', top: 0, right: 0 }}>
            <CloseIcon color={globals.styles.colors.colorBlack} />
          </ButtonOpacity>
          {/* Selected window box */}
          <View style={styles.selectorWindow} pointerEvents={'none'}>
            <Text>{'   '}</Text>
          </View>
          <PinkButton buttonStyle={styles.selectButton} handlePress={handleSelectTime} title={'SELECT'} />
        </View>
      ) : (
        <View />
      )}
    </Modal>
  )
}

export default TimePickerModal
