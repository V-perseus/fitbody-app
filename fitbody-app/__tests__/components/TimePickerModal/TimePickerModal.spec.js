import React from 'react'

import { render, cleanup, fireEvent } from '../../../testUtils'
import TimePickerModal from '../../../src/components/TimePickerModal'

describe('TimePickerModal', () => {
  afterEach(() => {
    cleanup()
  })

  it('renders snapshot correctly', () => {
    const tree = render(<TimePickerModal onChange={jest.fn} onClose={jest.fn} visible={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the default time option', () => {
    const { getByText } = render(<TimePickerModal onChange={jest.fn} onClose={jest.fn} visible={true} />)
    expect(getByText(/7/i)).not.toBeNull()
    expect(getByText(/00/gi)).not.toBeNull()
    expect(getByText(/am/gi)).not.toBeNull()
  })

  it('selects the current time option', async () => {
    const handleChange = jest.fn()
    const { getByText } = render(<TimePickerModal onChange={handleChange} onClose={jest.fn} visible={true} />)
    await fireEvent.press(getByText(/select/gi))
    expect(handleChange).toHaveBeenCalled()
    expect(handleChange).toHaveBeenCalledWith('07:00')
  })

  it('closes the modal', async () => {
    const handleClose = jest.fn()
    const { getByTestId, queryByText } = render(<TimePickerModal onChange={jest.fn} onClose={handleClose} visible={true} />)
    await fireEvent.press(getByTestId('close_btn'))
    expect(handleClose).toHaveBeenCalled()
    expect(queryByText(/select/gi)).toBeNull()
  })
})
