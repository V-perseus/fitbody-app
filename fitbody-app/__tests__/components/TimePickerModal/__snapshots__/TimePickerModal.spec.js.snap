// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePickerModal renders snapshot correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 10,
          "height": 435.1,
          "padding": 50,
          "paddingBottom": 0,
          "paddingTop": 20,
          "width": 675,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "width": 60,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fafafa",
                "flex": 1,
                "height": 345,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#ffffff",
                  "borderBottomWidth": 0.5,
                  "borderTopColor": "#ffffff",
                  "borderTopWidth": 0.5,
                  "height": 49.3,
                  "position": "absolute",
                  "top": 147.85,
                  "width": 750,
                }
              }
            />
            <RCTScrollView
              bounces={false}
              nestedScrollEnabled={true}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              showsVerticalScrollIndicator={false}
            >
              <View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": 147.85,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      1
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      2
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      3
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      4
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      5
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      6
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      7
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      8
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      9
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      10
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      11
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "black",
                            "fontFamily": "BebasNeue",
                            "fontSize": 24,
                            "marginBottom": 10,
                            "marginTop": 10,
                          },
                          Object {
                            "textAlign": "right",
                          },
                        ]
                      }
                    >
                      12
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": 147.85,
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "BebasNeue",
                "fontSize": 24,
                "marginBottom": 10,
                "marginTop": 10,
              },
              Object {
                "marginTop": 22,
                "textAlign": "right",
              },
            ]
          }
        >
          :
        </Text>
        <View
          style={
            Object {
              "width": 60,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fafafa",
                "flex": 1,
                "height": 345,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#ffffff",
                  "borderBottomWidth": 0.5,
                  "borderTopColor": "#ffffff",
                  "borderTopWidth": 0.5,
                  "height": 49.3,
                  "position": "absolute",
                  "top": 147.85,
                  "width": 750,
                }
              }
            />
            <RCTScrollView
              bounces={false}
              nestedScrollEnabled={true}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              showsVerticalScrollIndicator={false}
            >
              <View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": 147.85,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "BebasNeue",
                          "fontSize": 24,
                          "marginBottom": 10,
                          "marginTop": 10,
                        }
                      }
                    >
                      00
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "BebasNeue",
                          "fontSize": 24,
                          "marginBottom": 10,
                          "marginTop": 10,
                        }
                      }
                    >
                      30
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": 147.85,
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          style={
            Object {
              "width": 50,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fafafa",
                "flex": 1,
                "height": 345,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomColor": "#ffffff",
                  "borderBottomWidth": 0.5,
                  "borderTopColor": "#ffffff",
                  "borderTopWidth": 0.5,
                  "height": 49.3,
                  "position": "absolute",
                  "top": 147.85,
                  "width": 750,
                }
              }
            />
            <RCTScrollView
              bounces={false}
              nestedScrollEnabled={true}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              showsVerticalScrollIndicator={false}
            >
              <View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": 147.85,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "BebasNeue",
                          "fontSize": 24,
                          "marginBottom": 10,
                          "marginTop": 10,
                        }
                      }
                    >
                      AM
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 30,
                        "justifyContent": "center",
                      },
                      Object {
                        "height": 49.3,
                      },
                    ]
                  }
                >
                  <View>
                    <Text
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "BebasNeue",
                          "fontSize": 24,
                          "marginBottom": 10,
                          "marginTop": 10,
                        }
                      }
                    >
                      PM
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": 147.85,
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <BVLinearGradient
        colors={
          Array [
            4294967295,
            16777215,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        pointerEvents="none"
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 0,
            "height": 230,
            "position": "absolute",
            "top": 20,
            "width": 675,
          }
        }
      />
      <BVLinearGradient
        colors={
          Array [
            16777215,
            4294967295,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        pointerEvents="none"
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "borderRadius": 10,
            "borderWidth": 0,
            "bottom": 74,
            "height": 230,
            "position": "absolute",
            "width": 675,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={16}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "opacity": 1,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "padding": 15,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
        testID="close_btn"
      >
        <close.svg
          color="#333333"
        />
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#dedede",
            "borderTopWidth": 1,
            "height": 50,
            "position": "absolute",
            "top": 165.5,
            "width": 675,
          }
        }
      >
        <Text>
             
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 10,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "flexDirection": "column",
              "height": 55,
              "justifyContent": "center",
              "marginBottom": 10,
              "marginTop": 10,
              "opacity": 1,
              "width": 675,
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4294610389,
                4294610389,
              ]
            }
            endPoint={
              Object {
                "x": 0.5,
                "y": 1,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0.5,
                "y": 0,
              }
            }
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "height": 55,
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                  "width": 675,
                },
                Object {
                  "width": 600,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 14,
                  "letterSpacing": 0,
                }
              }
            >
              SELECT
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
